sensor/kubernetes/listener/watcher/crd/crdwatcher.go

```diff
@@ -84,33 +90,28 @@ func (w *crdWatcher) Watch(statusC chan<- *watcher.Status) error {
 				if !ok {
 					return
 				}
-				if !w.resources.Contains(event.resourceName) {
+				if !resources.Contains(event.resourceName) {
 					continue
 				}
 				switch event.action {
 				case central.ResourceAction_CREATE_RESOURCE:
-					w.availableResources.Add(event.resourceName)
+					availableResources.Add(event.resourceName)
 				case central.ResourceAction_REMOVE_RESOURCE:
-					w.availableResources.Remove(event.resourceName)
+					availableResources.Remove(event.resourceName)
 				}
 			}
-			var status *watcher.Status
-			if len(w.resources) == len(w.availableResources) && w.status.CompareAndSwap(false, true) {
-				status = &watcher.Status{}
-			}
-			if len(w.resources) > len(w.availableResources) && w.status.CompareAndSwap(true, false) {
-				status = &watcher.Status{}
-			}
-			if status != nil {
-				status.Available = w.status.Load()
-				status.Resources = w.resources
-				select {
-				case <-w.stopSig.Done():
-					return
-				case statusC <- status:
-				}
+			// Send status only when availability changes.
+			// If we reach cardinality and previous state was not available,
+			// or it was available but element was removed.
+			if (cardinality == len(availableResources) && !available) ||
+				(cardinality > len(availableResources) && available) {
+				available = !available
+				fn(&watcher.Status{
```

Comment:
Not sure if possible to go here:
```suggestion
				go fn(&watcher.Status{
```

Reply:
Why? this will create a gorutine while current implementation is not async as channel is unbuffered and we call callback withouth `go`

Reply:
No reason, probably.

Reply:
Unless we want to unblock the event sender eariler.

Reply:
but this creates new issues. how to stop this gorutine, is this calback meant for concurent runs 
I think this might be out of scope of this PR